Training/Training/harness.c

```diff
@@ -0,0 +1,127 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// Gokul TS
+// ------------------------------------------------------------------
+// harness.c
+// Program on main branch.
+// ------------------------------------------------------------------------------------------------
+#define _CRT_SECURE_NO_WARNINGS  1
+#include <windows.h>
+#include <stdio.h>
+#include <malloc.h>
+#include <io.h>
+
+/// <summary>
+/// This function will execute the FSM providing the input and output file names as arguments
+/// </summary>
+/// <param name="exeFilePathAndName"></param>
+/// <param name="inputFilePathAndName"></param>
+/// <param name="outputFilePathAndName"></param>
+/// <returns></returns>
+int ExecProgram (char* exeFilePathAndName, char* inputFilePathAndName, char* outputFilePathAndName) {
+   char* cmdline = malloc (strlen (exeFilePathAndName) + strlen (inputFilePathAndName) + strlen (outputFilePathAndName) + 3);
+   if (cmdline == NULL) {
+      printf ("Unable to allocate memory\n");
+      return 1;
+   }
+   sprintf (cmdline, "%s%s%s%s%s", exeFilePathAndName, " ", inputFilePathAndName, " ", outputFilePathAndName);
+
+   // Set up structures for process information
+   STARTUPINFOA si;
+   PROCESS_INFORMATION pi;
+
+   // Initialize memory for STARTUPINFO and PROCESS_INFORMATION structures
+   ZeroMemory (&si, sizeof (si));
+   si.cb = sizeof (si);
+   ZeroMemory (&pi, sizeof (pi));
+
+   // Attempt to create the process
+   if (!CreateProcessA (
+      NULL,            // Application name (NULL if using command line)
+      cmdline,          // Command line (path to the executable)
+      NULL,            // Process handle not inheritable
+      NULL,            // Thread handle not inheritable
+      FALSE,           // Set handle inheritance to FALSE
+      0,               // No creation flags
+      NULL,            // Use parent's environment block
+      NULL,            // Use parent's starting directory
+      &si,             // Pointer to STARTUPINFO structure
+      &pi))            // Pointer to PROCESS_INFORMATION structure
+   {
+      // If CreateProcess fails, print an error message
+      int err = GetLastError ();
+      printf ("Failed to start process. Error: %lu\n", GetLastError ());
+      free (cmdline);
+      return 1;
+   }
+
+   // Wait until the process has finished execution
+   WaitForSingleObject (pi.hProcess, INFINITE);
+
+   // Close handles
+   CloseHandle (pi.hProcess);
+   CloseHandle (pi.hThread);
+
+   free (cmdline);
+   return 0;
+
+}
+
+/// <summary> Returns whether the given two files are equal or not </summary>
+int CompareFiles (char* file, char* tempOutPath, int* errBit, int* errBitValue) {
```

Comment:
a) If you are just checking for equality or not, then it is preferable to return a boolean value rather than an int.
b) If you are checking for < or == or >, then return an integer value which can be checked for <0 or =0 or >0.
c) The name errBit does not convey the meaning i.e. that it is the bit number. The better name is errBitNo.